test/integration_test.py

```diff
@@ -559,12 +561,11 @@ def _assert_catalog_complete(self,
 
     def _assert_catalog_empty(self, catalog: CatalogName):
         with self.subTest('catalog_empty', catalog=catalog):
-            hit_counts = {
-                entity_type: len(self._get_entities(catalog, entity_type))
-                for entity_type in self.entity_types
-            }
-            log.info('Hit counts are %r', hit_counts)
-            self.assertEqual(0, sum(hit_counts.values()))
+            for entity_type in self.entity_types:
+                with self.subTest(entity_type=entity_type):
+                    hits = self._get_entities(catalog, entity_type)
+                    log.info('Hit counts for %s are %r', entity_type, hits)
```

Comment:
You're logging the hits array but the message calls that "hit counts". That's confusing.

Reply:
Besides, we don't need the log line. If the assert passes, we know that the list is empty. If it doesn't the hits are logged by the assert.